tests/scaletest/lib/parser.go

```diff
@@ -0,0 +1,227 @@
+	DisableStaticRouteSync string   `json:"disableStaticRouteSync"`
+	DefaultIngController   string   `json:"defaultIngController"`
+	NetworkName            string   `json:"NetworkName"`
+	VrfRefName             string   `json:"vrfRefName"`
+	Platform               Platform `json:"platform"`
+}
+
+func (cluster Cluster) GetPlatformDetails() Platform {
+	return cluster.Platform
+}
+
+func (cluster Cluster) GetClusterName() string {
+	return cluster.ClusterName
+}
+
+func (cluster Cluster) GetNetworkName() string {
+	return cluster.NetworkName
+}
+
+func (cluster Cluster) GetClusterId() string {
+	return cluster.ClusterId
+}
+
+func (cluster Cluster) GetKubeConfigFilePath() string {
+	return cluster.KubeConfigFilePath
+}
+
+func (cluster Cluster) GetCniPlugin() string {
+	return cluster.CniPlugin
+}
+
+func (cluster Cluster) GetCloudName() string {
+	return cluster.CloudName
+}
+
+func (cluster Cluster) GetDisableStaticRouteSync() string {
+	return cluster.DisableStaticRouteSync
+}
+
+func (cluster Cluster) GetDefaultIngController() string {
+	return cluster.DefaultIngController
+}
+
+func (cluster Cluster) GetVrfRefName() string {
+	return cluster.VrfRefName
+}
+
+type AkoParams struct {
+	NumClusters int       `json:"num_clusters"`
+	Clusters    []Cluster `json:"clusters"`
+}
+
+func (akoParams AkoParams) GetNumberOfClusters() int {
+	return akoParams.NumClusters
+}
+
+func (akoParams AkoParams) GetClusterList() []Cluster {
+	return akoParams.Clusters
+}
+
+func (akoParams AkoParams) GetCluster(clusterNum int) Cluster {
+	return akoParams.Clusters[clusterNum]
+}
+
+type Test struct{
+	Namespace 		  string `json:"namespace"`
+	AppName 		  string `json:"appName"`
+	ServiceNamePrefix string `json:"serviceNamePrefix"`
+	IngressNamePrefix string `json:"ingressNamePrefix"`
+}
+
+func (test Test) GetNamespace() string{
+	return test.Namespace
+}
+
+func (test Test) GetAppName() string{
+	return test.AppName
+}
+
+func (test Test) GetServiceNamePrefix() string{
+	return test.ServiceNamePrefix
+}
+
+func (test Test) GetIngressNamePrefix() string{
+	return test.IngressNamePrefix
+}
+
+type VMNetwork struct {
+	Management string `json:"mgmt"`
+}
+
+func (vmNet VMNetwork) GetNetworkManagement() string {
+	return vmNet.Management
+}
+
+type Controller struct {
+	DataCenter string    `json:"datacenter"`
+	Name       string    `json:"name"`
+	Cluster    string    `json:"cluster"`
+	Ip         string    `json:"ip"`
+	UserName   string 	 `json:"username"`
+	Password   string 	 `json:"password"`
+	Mask       string    `json:"mask"`
+	Network    VMNetwork `json:"networks"`
+	CloudName  string    `json:"cloud_name"`
+	Host       string    `json:"host"`
+	Static     string    `json:"static"`
+	Datastore  string    `json:"datastore"`
+	Type       string    `json:"type"`
+	Gateway    string    `json:"gateway"`
+	CpuCores   string    `json:"cpu_cores"`
+	PublicIp   string    `json:"public_ip"`
+}
+
+func (ctlr Controller) GetVMDataCenter() string {
```

Comment:
These public variables can be accessed by objectname.variable name also. Defining another get function may not be necessary.